source/manual/architecture-deep-dive.html.md

```diff
@@ -0,0 +1,462 @@
+layers. Some requests make it through to origin, where they're routed to the
+machine running the (usually Rails-based) rendering application that knows how
+to handle the request.
+
+## What happens when someone hits 'Publish'?
+
+### Draft and live stacks
+
+![](images/draft-live-stacks.png)
+
+Everything you've just read about in the first section exists in two stacks:
+draft and live. These are very similar to each other: each is a collection
+of machines in the cloud, running GOV.UK applications. Everything that runs in
+the live stack also runs in the draft stack, in order to have a way of
+previewing content in a non-public-facing way. However, the draft stack also
+has additional machines that run the [publishing apps].
+
+Applications shouldn't know what stack they're in - they're simply configured
+to talk to other applications in their stack.
+
+The live stack entry point is the 'router' app. You can swap `www` for
+`www-origin` to bypass Fastly and view the live stack at origin. This is
+only available to office IPs / VPN, and to Fastly IP addresses (configured in
+[govuk-provisioning]).
+
+The draft stack entry point is [authenticating-proxy], which sits in front
+of 'router'. You can swap `www` for `draft-origin` to view the draft stack at
+origin. The draft stack is not IP-restricted, as we need to be able to share
+links to be reviewed ("2i'd") or fact-checked by non-Government departments.
+It is, however, only visible to users who have been verified through
+authenticating-proxy, by signing into [signon] (an authentication and
+authorisation portal) or by providing a valid `auth_bypass_id` (as a URI
+parameter or session cookie). Read more about previews in ["How the draft stack works"].
+
+Signon doubles up as an authorisation platform, as it associates users with
+arbitrary permissions, so a publishing app can query if the current user has
+the necessary permissions to perform a given action, such as publishing
+content.
+
+[authenticating-proxy]: https://github.com/alphagov/authenticating-proxy
+[govuk-provisioning]: https://github.com/alphagov/govuk-provisioning
+["How the draft stack works"]: /manual/content-preview.html
+[publishing apps]: /#publishing-apps
+[signon]: /apps/signon.html
+
+### Publishing API vs Content Store
+
+![](images/publishing-api-content-store.png)
+
+At this point it's probably worth summarising what "content" is. Almost every
+piece of content on GOV.UK lives in a database called the "[content store]",
+which stores only the latest "edition" of that content. Internally the content
+is referred to as a "document", even if it is not itself a document.
+
+Content is published to the content store via the [publishing API], which
+stores all of the editions of the document, and performs validation checks
+whenever it receives a new edition. Every piece of content has a `document_type`
+corresponding to a particular JSON schema defined in [govuk-content-schemas].
```

Comment:
The relevant schema is decided by the `schema_name`, not the `document_type`.

Reply:
Oops, good spot! 👏